src/AiClient.php

```diff
@@ -0,0 +1,413 @@
+    {
+        throw new \RuntimeException(
+            'MessageBuilder is not yet available. This method depends on builder infrastructure. ' .
+            'Use direct generation methods (generateTextResult, generateImageResult, etc.) for now.'
+        );
+    }
+
+    /**
+     * Template method for executing generation operations.
+     *
+     * @since n.e.x.t
+     *
+     * @param string|MessagePart|MessagePart[]|Message|Message[] $prompt The prompt content.
+     * @param ModelInterface|null $model Optional specific model to use.
+     * @param string $type The generation type.
+     * @return GenerativeAiResult The generation result.
+     *
+     * @throws \InvalidArgumentException If the prompt format is invalid.
+     * @throws \RuntimeException If no suitable model is found.
+     */
+    private static function executeGeneration($prompt, ?ModelInterface $model, string $type): GenerativeAiResult
+    {
+        // Convert prompt to standardized Message array format
+        $messages = PromptNormalizer::normalize($prompt);
+        /** @var list<Message> $messageList */
+        $messageList = array_values($messages);
+
+        // Map type to specific methods
+        switch ($type) {
+            case 'text':
+                $resolvedModel = $model ?? ModelDiscovery::findTextModel(self::defaultRegistry());
+                InterfaceValidator::validateTextGeneration($resolvedModel);
+                /** @phpstan-ignore-next-line */
+                return $resolvedModel->generateTextResult($messageList);
+
+            case 'image':
+                $resolvedModel = $model ?? ModelDiscovery::findImageModel(self::defaultRegistry());
+                InterfaceValidator::validateImageGeneration($resolvedModel);
+                /** @phpstan-ignore-next-line */
+                return $resolvedModel->generateImageResult($messageList);
+
+            case 'speech':
+                $resolvedModel = $model ?? ModelDiscovery::findSpeechModel(self::defaultRegistry());
+                InterfaceValidator::validateSpeechGeneration($resolvedModel);
+                /** @phpstan-ignore-next-line */
+                return $resolvedModel->generateSpeechResult($messageList);
+
+            default:
+                throw new \InvalidArgumentException("Unsupported generation type: {$type}");
+        }
+    }
+
+    /**
+     * Generates text using the traditional API approach.
+     *
+     * @since n.e.x.t
+     *
+     * @param string|MessagePart|MessagePart[]|Message|Message[] $prompt The prompt content.
+     * @param ModelInterface|null $model Optional specific model to use.
+     * @return GenerativeAiResult The generation result.
+     *
+     * @throws \InvalidArgumentException If the prompt format is invalid.
+     * @throws \RuntimeException If no suitable model is found.
+     */
+    public static function generateTextResult($prompt, ModelInterface $model = null): GenerativeAiResult
```

Comment:
`$model` type must be nullable if it can default to null.
```suggestion
    public static function generateTextResult($prompt, ?ModelInterface $model = null): GenerativeAiResult
```

Reply:
resolved https://github.com/WordPress/php-ai-client/pull/55/commits/ce4c14a8c6ff7dace95cc6976dbd8a334899b659